### PR TITLE
Automated PR: Cookstyle Changes

### PR DESCRIPTION
**Pull request by xorimabot**
Hey!
I ran Cookstyle 7.12.2 against this repo and here are the results.
This repo was selected due to the topics of chef-cookbook

### Issues found and resolved with resources/mysql_database.rb

 - 1:1 refactor: `Chef/Deprecations/ResourceWithoutUnifiedTrue` - Set `unified_mode true` in Chef Infra Client 15.3+ custom resources to ensure they work correctly in Chef Infra Client 18 (April 2022) when Unified Mode becomes the default. (https://docs.chef.io/workstation/cookstyle/chef_deprecations_resourcewithoutunifiedtrue)

### Issues found and resolved with resources/mysql_user.rb

 - 1:1 refactor: `Chef/Deprecations/ResourceWithoutUnifiedTrue` - Set `unified_mode true` in Chef Infra Client 15.3+ custom resources to ensure they work correctly in Chef Infra Client 18 (April 2022) when Unified Mode becomes the default. (https://docs.chef.io/workstation/cookstyle/chef_deprecations_resourcewithoutunifiedtrue)

### CODE REVIEW

**Comment by damacus** [comment]
Chef version also needs setting to 15.3

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
This needs the selinux cookbook to be updated first to support `unified_mode`.

File: libraries/mysql_service_base.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def create_config
         # setting up multiple services.
         file "#{prefix_dir}/etc/mysql/my.cnf" do
           action :delete
-          not_if { node['init_package'] == 'systemd' }
+          not_if { systemd? }
```
Comment (tas50) [inline libraries/mysql_service_base.rb:44]: This means it requires 15.5+

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [11.0.0](https://github.com/sous-chefs/mysql/releases/tag/11.0.0)

**Pull request merged**